prowler/lib/outputs/finding.py

```diff
@@ -267,3 +274,183 @@ def generate_output(
                 f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
             raise error
+
+    @classmethod
+    def transform_api_finding(cls, finding) -> "Finding":
+        """
+        Transform a FindingModel instance into an API-friendly Finding object.
+
+        This class method extracts data from a FindingModel instance and maps its
+        properties to a new Finding object. The transformation populates various
+        fields including authentication details, timestamp, account information,
+        check metadata (such as provider, check ID, title, type, service, severity,
+        and remediation details), as well as resource-specific data. The resulting
+        Finding object is structured for use in API responses or further processing.
+
+        Args:
+            finding (API Finding): An API Finding instance containing data from the database.
+
+        Returns:
+            Finding: A new Finding instance populated with data from the provided model.
+        """
+        output_data = {}
+        output_data["auth_method"] = ""
+        output_data["timestamp"] = finding.inserted_at
+        output_data["account_uid"] = finding.scan.provider.uid
+        output_data["account_name"] = ""
```

Comment:
@jfagoagas review

Reply:
- GCP -> Nothing, so empty string
- Azure -> `{azure_provider.region_config.name}`
- K8s -> Nothing, so empty string
- AWS -> `{aws_provider.identity.partition}`

I think you should use `get_nested_attribute` function to get all of the above as it is done in the `Finding` class, e.g.: `get_nested_attribute(gcp_provider, "identity.profile")` since it defaults to `""` if not present.

Reply:
I'm thinking that if we can pass the `provider` object to this function we could get also:
- `account_name`
- `account_email`
- `account_organization_uid`
- `account_organization_name`
- `account_tags`

for the different providers as it is done in the `generate_output` class method.

Reply:
Yes it is included